### ISSUE TITLE
Rasa model versions backwards compatibility 

### ISSUE THREAD

**Issue by Ben-Mark** [RasaHQ/rasa#5799]
**Description of Problem**:
After rasa-x / rasa version upgrade , previous models from previous versions are incompatible.
if it was once in a year, could be ok, but after every version (every few days) it's becoming a problem.
look at the tag's release date, its every 3/4 days :
https://hub.docker.com/r/rasa/rasa-x/tags
https://hub.docker.com/r/rasa/rasa/tags

**Overview of the Solution**:
backwards compatibility!

**Examples** (if relevant):
the compatibility matrix, this is crazy!
https://rasa.com/docs/rasa-x/changelog/compatibility-matrix/

**Blockers** (if relevant):
every rasa upgrade, is a complete mess/blockers.

**Definition of Done**:
enhance backwards compatibility

**Comment by sara-tagger** [comment]
Thanks for submitting this feature request 🚀 @tttthomasssss will get back to you about it soon! ✨

**Comment by tttthomasssss** [comment]
Hi @Ben-Mark, thanks very much for your raising this issue.

We're trying to avoid model breaking changes as much as possible - aiming to keep them at once a month. Can you describe what specifically bothers you? Is it the re-training of the models? Are you auto-pulling on latest?

rasa x is still pre-1.0 and there are lots of improvements happening at the moment.

Let me know if you have any further questions.

**Comment by Ben-Mark** [comment]
We're storing the trained models in s3
Trained models from rasa version 1.9.7
are not compatible with  rasa version 1.10.0

But last 2 weeks we've upgraded rasa from 1.8.3 to 1.9.7 and we had to re train everything.
We a large number of different rasa nlu data in separated rasa instances.

on v 1.10.0 it requires us to re train everything again , this is a pain.

I believe when upgrading a rasa version , allowing models to be  backwards compatible is essential.
Hope you do so 2.
